crates/services/p2p/src/discovery/dnsaddr_resolution.rs

```diff
@@ -0,0 +1,101 @@
+use anyhow::anyhow;
+use hickory_resolver::TokioAsyncResolver;
+use libp2p::Multiaddr;
+use std::pin::Pin;
+
+/// The prefix for `dnsaddr` protocol TXT record lookups.
+const DNSADDR_PREFIX: &str = "_dnsaddr.";
+/// The maximum number of DNS lookups when dialing.
+/// This limit is for preventing malicious or misconfigured DNS records from causing infinite recursion.
+const MAX_DNS_LOOKUPS: usize = 10;
+
+#[async_trait::async_trait]
```

Comment:
Async traits have been stabilized since [rust 1.75](https://blog.rust-lang.org/2023/12/28/Rust-1.75.0.html), so I'd prefer not to add this declaration for new traits. Instead just have the functions return `impl Future<Output = ...>` and use `async fn` in the implementations.

Reply:
removed the trait